ocamldebug.el

```diff
@@ -717,8 +718,48 @@ Obeying it means displaying in another window the specified file and line."
 
 ;;; Miscellaneous.
 
+(defun ocamldebug--read-from-file (file)
+  "Read FILE as a list of sexps. If invalid syntax, return nil and message the error."
+  (with-temp-buffer
+    (save-excursion
+      (insert "(\n")
+      (insert-file-contents file)
+      (goto-char (point-max))
+      (insert "\n)\n"))
+    (condition-case err
+        (read (current-buffer))
+      ((error err) (progn (message "Error reading file %S: %S" file err) nil)))))
+
+(defun ocamldebug--find-single-library (sexps)
+  "If list SEXPS has a single element whose `car' is \"library\", return it.
+Otherwise return `nil'."
+  (let ((libs (seq-filter (lambda (elt) (equal (car elt) 'library)) sexps)))
+    (and (null (cdr libs)) (car libs))))
+
+(defun ocamldebug--dune-library-name (lib)
+  "With LIB a dune-syntax library stanza, get its name as a string."
+  (let ((field (or (seq-find (lambda (field) (and (consp field) (equal (car field) 'name))) lib)
+                   (seq-find (lambda (field) (and (consp field) (equal (car field) 'public\_name))) lib))))
+    (symbol-name (car (cdr field)))))
```

Comment:
Instead of `(and (consp field) (equal (car field) 'name))` you can just use `(eq (car-safe field) 'name)`.
You might also want to check that `field` is non-nil.
And please try to stay within 80 columns.

Reply:
> You might also want to check that field is non-nil.

Isn't that implied by (eq (car-safe field) 'name)?

Reply:
Sorry my comment was unclear indeed.
I meant the `field` that's let-bound (i.e. test before calling `symbol-name`) not the one(s) that's lambda-bound.

Reply:
I think the file would be malformed if there is neither `name` or `public_name` in a `(library)`

Reply:
Still, returning the string `"nil"` in that case will probably lead to puzzling behavior, no?
Shouldn't we instead emit a more "upfront" error helping the user fix his file?